src/helpers.jl

```diff
@@ -0,0 +1,407 @@
+    (p.Nq2 == 0 || p.Nq3 == 0 || p.Nj == 0) && return false
+    true
+end
+isRhadron(p::Union{Particle, Integer}) = isRhadron(convert(PDGID, p))
+
+
+"""
+$(SIGNATURES)
+
+Fundamental SUSY particles have N = 1 or 2.
+"""
+function isSUSY(p::PDGID)
+    !isstandard(p) && return false
+    p.N != 1 && p.N != 2 && return false
+    p.Nr != 0 && return false
+    fundamentalid(p) == 0 && return false
+    true
+end
+isSUSY(p::Union{Particle, Integer}) = isSUSY(convert(PDGID, p))
+
+
+"""
+$(SIGNATURES)
+
+Pentaquark IDs are of the form +/- 9 Nr Nl Nq1 Nq2 Nq3 Nj,
+where Nj = 2J + 1 gives the spin and Nr Nl Nq1 Nq2 Nq3 denote the quark numbers
+in order Nr >= Nl >= Nq1 >= Nq2 and Nq3 gives the antiquark number.
+"""
+function ispentaquark(p::PDGID)
+    !isstandard(p) && return false
+    p.N != 9 && return false
+    (p.N == 9 || p.Nr == 0) && return false
+    (p.Nj == 9 || p.Nl == 0) && return false
+    p.Nq1 == 0 && return false
+    p.Nq2 == 0 && return false
+    p.Nq3 == 0 && return false
+    p.Nj == 0 && return false
+    p.Nq2 > p.Nq1 && return false
+    p.Nq1 > p.Nl && return false
+    p.Nl > p.Nr && return false
+    true
+end
+ispentaquark(p::Union{Particle, Integer}) = ispentaquark(convert(PDGID, p))
+
+isgaugebosonorhiggs(p::PDGID) = 21 <= abs(p.value) <= 40
+isgaugebosonorhiggs(p::Union{Particle, Integer}) = isgaugebosonorhiggs(convert(PDGID, p))
+issmgaugebosonorhiggs(p::PDGID) = abs(p.value) == 24 || 21 <= p.value <= 25
+issmgaugebosonorhiggs(p::Union{Particle, Integer}) = issmgaugebosonorhiggs(convert(PDGID, p))
+
+function istechnicolor(p::PDGID)
+    !isstandard(p) && return false
+    p.N == 3
+end
+istechnicolor(p::Union{Particle, Integer}) = istechnicolor(convert(PDGID, p))
+
+"""
+$(SIGNATURES)
+
+Excited (composite) quarks and leptons have N = 4 and Nr = 0.
+"""
+function iscompositequarkorlepton(p::PDGID)
+    !isstandard(p) && return false
+    fundamentalid(p) == 0 && return false
+    !(p.N == 4 && p.Nr == 0) && return false
+    true
+end
+iscompositequarkorlepton(p::Union{Particle, Integer}) = iscompositequarkorlepton(convert(PDGID, p))
+
+"""
+$(SIGNATURES)
+
+Magnetic monopoles and Dyons are assumed to have one unit of Dirac monopole
+charge and a variable integer number xyz units of electric charge, where xyz
+stands for Nq1 Nq2 Nq3.
+Codes 411xyz0 are used when the magnetic and electrical charge sign agree and
+412xyz0 when they disagree, with the overall sign of the particle set by the
+magnetic charge. For now, no spin information is provided.
+"""
+function isdyon(p::PDGID)
+    !isstandard(p) && return false
+    p.N != 4 && return false
+    p.Nr != 1 && return false
+    !(p.Nl ∈ [1, 2]) && return false
+    p.Nq3 == 0 && return false
+    p.Nj != 0 && return false
+    true
+end
+isdyon(p::Union{Particle, Integer}) = isdyon(convert(PDGID, p))
+
+function isdiquark(p::PDGID)
+    !isstandard(p) && return false
+    abs(p.value) <= 100 && return false
+    0 < fundamentalid(p) <= 100 && return false
```

Comment:
How about `!isfundamental(p) && return false` instead of 202-204?

Reply:
That's not equivalent. In fact, it would not identify any of the diquarks.

I'd rather stick to the original implementation and logic instead of being smart here; it's also much easier to compare both packages, that's why I have not spent time on simplify things.

Also note that performance might be better with the step-wise approach due to shortcuts.